### PR TITLE
Clean Up Regional Partner Guardrail Field Changes

### PR DESCRIPTION
**Pull request by TurnerRiley**
This PR responds to comments made on [this PR](https://github.com/code-dot-org/code-dot-org/pull/49318) and cleans up the changes made there and the surrounding code. That PR was merged into `staging` and was urgent so these changes weren't made previously in case it slowed down its deploy. This PR will just be merged into `staging-next` since most of the changes are for readability and consistency.

Changes made:

- On the Regional Partners page, flip ordering of URG and FRL fields here since FRL appears before URG in the other spots they are listed.
- For the application scholarship requirements calculations, move hard-coded values (such as '50%' into default constants).
- Update test names to be clearer about desired behavior

Here's the behavior from [the previous PR](https://github.com/code-dot-org/code-dot-org/pull/49318) and this one to show it is still consistent despite the changes:

### Edit Page
Previous:
Edit_Page
New:
FRL now on top EDIT

## Show Page
Previous:
Show_Page
New:
FRL now on top SHOW

### If set to 'No Partner' for RP (default to 50% for both guardrails)
Previous:
No_Partner
New:
No_Partner

### If assigned RP (in this case, Reggie Partner), but RP has not explicitly set guardrails (default to 50% for both guardrails)
Previous:
Partner_not_set
New:
Partner_not_set

### If assigned RP (in this case, Reggie Partner, and RP has set guardrails (URG - 48%, FRL - 55%)
Previous:
Partner_set
New:
Partner_set

## Links
PR with original changes: [here](https://github.com/code-dot-org/code-dot-org/pull/49318)
Jira ticket: [here](https://codedotorg.atlassian.net/browse/ACQ-266?atlOrigin=eyJpIjoiMjA1YmNmZDk1N2Q4NGFhYTg1NDU2MzNkNWJhNDQ2YjgiLCJwIjoiaiJ9)

## Testing story
Local testing to ensure same behavior and drone tests.

## PR Checklist:
- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/models/pd/application/teacher_application_test.rb [hunk 1]
```diff
@@ -869,7 +869,7 @@ class TeacherApplicationTest < ActiveSupport::TestCase
       assert_equal(YES, application.response_scores_hash[:meets_scholarship_criteria_scores][:underrepresented_minority_percent])
     end
 
-    test 'scholarship criteria for rp guardrail percents rp did not specify guardrails' do
+    test 'scholarship criteria uses default guardrails when regaionl partner does not specify' do
```
Comment (megcrenshaw) [inline dashboard/test/models/pd/application/teacher_application_test.rb]: nit: `regional`

File: dashboard/test/models/pd/application/teacher_application_test.rb [hunk 2]
```diff
@@ -849,7 +849,7 @@ class TeacherApplicationTest < ActiveSupport::TestCase
       assert application.reload.principal_approval_state.include? 'Complete - Admin said Yes on'
     end
 
-    test 'scholarship criteria for rp guardrail percents rp set fields' do
+    test 'scholarship criteria uses regional partner set fields when specified' do
       regional_partner = build :regional_partner, frl_guardrail_percent: 52, urg_guardrail_percent: 48
```
Comment (megcrenshaw) [inline dashboard/test/models/pd/application/teacher_application_test.rb]: Nit: could use constants + / - here too

**Review by megcrenshaw — APPROVED** [review]

File: dashboard/test/models/pd/application/teacher_application_test.rb (hunk 1 above)
Comment (TurnerRiley) [reply dashboard/test/models/pd/application/teacher_application_test.rb]: Ack, that's embarrassing! Thanks for catching that!

Comment (megcrenshaw) [reply dashboard/test/models/pd/application/teacher_application_test.rb]: I've done things much more embarrassing, that then get merged. So no worries

**Pull request merged**